Tests/SentryTests/SentryCrash/SentryCrashDynamicLinkerTests.m

```diff
@@ -0,0 +1,154 @@
+#import "SentryCrashDynamicLinker.h"
+#import <XCTest/XCTest.h>
+#import <mach-o/dyld.h>
+#import <mach-o/dyld_images.h>
+#if TARGET_OS_IOS
+#    import <UIKit/UIKit.h>
+#endif
+#import <Foundation/Foundation.h>
+
+// Added for tests
+extern void sentrycrashdl_clearDyld(void);
+struct dyld_all_image_infos *getAllImageInfo(void);
+extern uint32_t imageIndexContainingAddress(const uintptr_t address);
+extern bool sentrycrashbic_shouldAddDyld(void);
+extern uintptr_t firstCmdAfterHeader(const struct mach_header *const header);
+#if __LP64__
+#    define SEGMENT_TYPE LC_SEGMENT_64
+#    define segment_command_t struct segment_command_64
+#else
+#    define SEGMENT_TYPE LC_SEGMENT
+#    define segment_command_t struct segment_command
+#endif
```

Comment:
I think we could at least expose some of this in the implementation's header, or a +Test.h header so we don't have to worry about the definitions drifting